### DISCUSSION TITLE
How can i tell if any of the layers are loading?

### DISCUSSION THREAD

**Discussion by paulleonartcalvo** [visgl/deck.gl#8213] [Q&A]
I am using deck.gl in combination with React Map GL. I have a memoized function which returns the layers (MVT) from a getMVTLayer function. That function takes some props and returns the mvt layer. 

```
const deckLayers = useMemo(
    () => [
      ...layers.map((layer) =>
        getMVTLayerFromLayer(
          layer,
          DatawayDatasourceSchemasMap[layer.datasourceId],
        )
      )
    ],
    [layers]
  );
```

I am trying to show a MUI `<CircularProgress/>` component while any of the layers are loading async data. I tried by doing the following: 

```
const layersAreLoading = deckLayers.some((l) => !l.isLoaded);
```

It does not seem to cause the loading spinner to disappear even after the tiles have clearly stopped loading. What am i missing?

**Comment by Pessimistress** [comment]
It's not clear how you are using `layersAreLoading`. You need to recompute it to get the latest loading state.

You can either use `setInterval`, or trigger a rerender upon DeckGL's `onAfterRender`.

**Reply by paulleonartcalvo** [reply]
Sorry for the lack of clarity, i'm using it in a conditional render in a component that render a react map GL map with an overlay which contained a spinner.  Like this:

```

const deckLayers = useMemo(
    () => [
      ...layers.map((layer) =>
        getMVTLayerFromLayer(
          layer,
          DatawayDatasourceSchemasMap[layer.datasourceId],
        )
      )
    ],
    [layers]
  );

const layersAreLoading = deckLayers.some((l) => !l.isLoaded);

return <MyComponent>
    {layersAreLoading && <CircularProgress />}
    <DeckGL {...} layers={deckLayers} >
        <ReactMapGLMap {...} />
    </DeckGL>
</MyComponent

```

And yes, the struggle appears to stem from the fact that i am not recomputing the `layersAreLoading` often enough. What is the recommended approach to achieve this behavior in a "react" way? Is forcing a re-render `onAfterRender` it? Thank you for the help!

**Comment by plgagnon2** [comment]
I am struggling on something similar but with https://deck.gl/docs/api-reference/google-maps/google-maps-overlay in a non-react way

I have some TileLayer https://deck.gl/docs/api-reference/geo-layers/tile-layer in it

I want to add a loading indicator when tiles starts loading and remove it when they are all loaded. I see some callbacks where I can handle the "loaded" state (https://deck.gl/docs/api-reference/geo-layers/tile-layer#callbacks) but I cant find a way to display the indicator when tiles are currently being fetched

**Reply by plgagnon2** [reply]
I managed like this 
```
new CustomTileLayer({
    ...
    data,
    onTilesLoading: () => setTilesLoading(true),
    onViewportLoad: () => setTilesLoading(false)
});

interface CustomTileLayerProps {
    onTilesLoading?: () => void;
}

class CustomTileLayer<DataT> extends TileLayer<DataT, CustomTileLayerProps> {
    getTileData(tile: TileLoadProps) {
        const {
            props: { onTilesLoading }
        } = this;
        onTilesLoading?.();

        return super.getTileData(tile) ;
    }
}
```
